Multiphysic DEM - Heated Packed Bed example (#1561)
chaos-polymtl/lethe: doc/source/examples/dem-mp/heated-packed-bed/heated-packed-bed.rst

      set surface roughness wall       = 0.1e-9
      set thermal accommodation wall   = 0.7
      set thermal conductivity gas     = 0.027
      set specific heat gas            = 1006
      set dynamic viscosity gas        = 1.85e-5
      set specific heats ratio gas     = 1
      set molecular mean free path gas = 68.e-9
    end

Model Parameters
~~~~~~~~~~~~~~~~

For the first two stages, the model parameters are defined as:

.. code-block:: text

    subsection model parameters
      subsection contact detection
        set contact detection method                = dynamic
        set dynamic contact search size coefficient = 0.9
        set neighborhood threshold                  = 1.3
      end
      subsection load balancing
        set load balance method = frequent
        set frequency           = 100000
      end
      set particle particle contact force method = hertz_mindlin_limit_overlap
      set rolling resistance torque method       = constant_resistance
      set particle wall contact force method     = nonlinear
      set integration method                     = velocity_verlet
      set solver type                            = dem_mp
    end

> For the heating of the particles, the parameter ``disable position integration`` is set to ``true`` to freeze the position of the particles. This allows to use a higher time step for the evolution of the temperature. As particles are not moving, ``load balancing`` is no longer necessary.

Reviewer: ```suggestion
For the heating of the particles, the parameter ``disable position integration`` is set to ``true`` to freeze the position of the particles by disabling the time-integration of the particle velocity and position. This allows to use a higher time step for the evolution of the temperature since the collisions are not integrated in time anymore. As particles are not moving, ``load balancing`` is no longer necessary.
```

.. code-block:: text

    subsection model parameters
      subsection contact detection
        set contact detection method                = dynamic
        set dynamic contact search size coefficient = 0.9
        set neighborhood threshold                  = 1.3
      end
      set particle particle contact force method = hertz_mindlin_limit_overlap
      set rolling resistance torque method       = constant_resistance
      set particle wall contact force method     = nonlinear
      set integration method                     = velocity_verlet
      set solver type                            = dem_mp
      set disable position integration           = true
    end

Solid Objects
~~~~~~~~~~~~~~~

> Three solid surfaces are used in this example. The first one is the one used to heat the packed bed from :math:`7` s to :math:`4000` s, with a temperature of :math:`53°C`. The second one is used to even the particles on the side of the packed bed. The last one closes the box to maintain the particles within it when the direction of the gravity is changed. The last two walls are both set to ``adiabatic``, meaning that they are not conductive.

Reviewer: ```suggestion
Three solid surfaces are used in this example. The first one is the one used to heat the packed bed from :math:`7` s to :math:`4000` s, with a temperature of :math:`53°C`. The second one is used to even the particles on the side of the packed bed. The last one closes the box to maintain the particles within it when the direction of the gravity is changed. The last two walls are both set to ``adiabatic``, meaning that are insulated and do not provide any heat transfer.
```

    :align: center

The results show good agreement with the experimental and numerical results of Beaulieu but still undershoot a bit the temperature at :math:`h_1 = 4.0` cm compared to the experimental data. Our temperature curves are slightly higher than the ones numerically obtained by Beaulieu due to using the hertz contact radius instead of the analytical one. The higher Young's modulus also gives better results.

It has been noticed while trying different methods to load the particles that the results change a lot according to the loading method but also depending on the walls friction coefficient.
The following figure shows the results of the simulation with the ``friction coefficient wall`` set to :math:`1.0` instead of :math:`0.7`.

.. figure:: images/mean-temperatures-friction.png
    :width: 500
    :align: center

This ``friction coefficient wall`` allows to fit the experimental data better but it is debatable whether a friction coefficient of :math:`1.0` is realistic to model the Styrofoam walls used in the experiment. Also, other parameters like the ``rolling friction`` should have probably been adjusted as well to fit the experimental properties.

---------

> Reference

Reviewer: You could add a possibilities for extension and mention that there are other particles type and other results could be reproduced.

      set surface roughness wall       = 0.1e-9
      set thermal accommodation wall   = 0.7
      set thermal conductivity gas     = 0.027
      set specific heat gas            = 1006
      set dynamic viscosity gas        = 1.85e-5
      set specific heats ratio gas     = 1
      set molecular mean free path gas = 68.e-9
    end

Model Parameters
~~~~~~~~~~~~~~~~

For the first two stages, the model parameters are defined as:

.. code-block:: text

    subsection model parameters
      subsection contact detection
        set contact detection method                = dynamic
        set dynamic contact search size coefficient = 0.9
        set neighborhood threshold                  = 1.3
      end
      subsection load balancing
        set load balance method = frequent
        set frequency           = 100000
      end
      set particle particle contact force method = hertz_mindlin_limit_overlap
      set rolling resistance torque method       = constant_resistance
      set particle wall contact force method     = nonlinear
      set integration method                     = velocity_verlet
      set solver type                            = dem_mp
    end

> For the heating of the particles, the parameter ``disable position integration`` is set to ``true`` to freeze the position of the particles by disabling the time-integration of the particle velocity and position. This allows to use a higher time step for the evolution of the temperature since the collisions are not integrated in time anymore. As particles are not moving, ``load balancing`` is no longer necessary.

Reviewer 2: ```suggestion
For the heating phase of the particles, the parameter ``disable position integration`` is set to ``true`` to freeze the particles in place by disabling the time-integration of the particles velocity and position. This allows for the use of a larger time step for the evolution of the temperature, as collisions no longer need to be resolved. ASince the particles remain stationary, ``load balancing`` is also no longer necessary.
```

.. code-block:: text

    subsection model parameters
      subsection contact detection
        set contact detection method                = dynamic
        set dynamic contact search size coefficient = 0.9
        set neighborhood threshold                  = 1.3
      end
      set particle particle contact force method = hertz_mindlin_limit_overlap
      set rolling resistance torque method       = constant_resistance
      set particle wall contact force method     = nonlinear
      set integration method                     = velocity_verlet
      set solver type                            = dem_mp
      set disable position integration           = true
    end

Solid Objects
~~~~~~~~~~~~~~~

> Three solid surfaces are used in this example. The first one is the one used to heat the packed bed from :math:`7` s to :math:`4000` s, with a temperature of :math:`53°C`. The second one is used to even the particles on the side of the packed bed. The last one closes the box to maintain the particles within it when the direction of the gravity is changed. The last two walls are both set to ``adiabatic``, meaning that they are insulated and do not provide any heat transfer.

Reviewer 2: ```suggestion
Three solid surfaces are used in this example. The first one is the one used to heat the packed bed from :math:`7` s to :math:`4000` s, with a temperature of :math:`53°C`. The second one is used to even the particles on the side of the packed bed. The last one closes the box to maintain the particles within it when the direction of the gravity is changed. The second and third walls are both set to ``adiabatic``, meaning that they are insulated and do not provide any heat transfer.
```

---------------

A Python post-processing code ``bed-postprocessing.py`` is provided with this example. It is used to compare the temperature of the packed-bed at three different heights :math:`h_1 = 4.0` cm, :math:`h_2 = 6.0` cm and :math:`h_3 = 7.3` cm (:math:`h = 0.0` cm corresponds to the top wall), with the results obtained experimentally and numerically by Beaulieu for stainless steel.

.. figure:: images/heights.png
    :height: 400
    :align: center

The post-processing code can be run with the following command. The argument is the folder which contains the ``.prm`` file.

.. code-block:: text
  :class: copy-button

    python3 bed-postprocessing.py  --folder ./

.. important::

    You need to ensure that ``lethe_pyvista_tools`` is working on your machine. Click `here <../../../tools/postprocessing/postprocessing_pyvista.html>`_ for details.

-------
Results
-------

The following figure compares the temperature of the packed-bed at three different heights :math:`h_1 = 4.0` cm, :math:`h_2 = 6.0` cm and :math:`h_3 = 7.3` cm, with the results obtained by Beaulieu for stainless steel.

.. figure:: images/mean-temperatures.png
    :width: 500
    :align: center

> The results show good agreement with the experimental and numerical results of Beaulieu but still undershoot a bit the temperature at :math:`h_1 = 4.0` cm compared to the experimental data. Our temperature curves are slightly higher than the ones numerically obtained by Beaulieu due to using the hertz contact radius instead of the analytical one. The higher Young's modulus also gives better results.

Reviewer 2: ```suggestion
The results show good agreement with the experimental and numerical results of Beaulieu but still undershoot a bit the temperature at :math:`h_1 = 4.0` cm compared to the experimental data. Our temperature curves are slightly higher than the ones numerically obtained by Beaulieu due to the hertz contact radius model used instead of the analytical one as in Beaulieu's work. The higher Young's modulus employed also gives better results.
```

> It has been noticed while trying different methods to load the particles that the results change a lot according to the loading method but also depending on the walls friction coefficient.
> The following figure shows the results of the simulation with the ``friction coefficient wall`` set to :math:`1.0` instead of :math:`0.7`.

Reviewer 2: ```suggestion
It has been noticed while expermienting with the model that the simulation results are highly sensitive to the loading method and to the wall friction coefficient.
The following figure shows the results of the simulation with the ``friction coefficient wall`` set to :math:`1.0` instead of :math:`0.7`.
```
